os/include/tinyara/mpu.h

```diff
@@ -0,0 +1,53 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+///@file tinyara/mpu.h
+
+#ifndef __INCLUDE_TINYARA_MPU_H
+#define __INCLUDE_TINYARA_MPU_H
+
+/********************************************************************************
+ * Included Files
+ ********************************************************************************/
+
+#include <tinyara/config.h>
+
+/********************************************************************************
+ * Pre-processor Definitions
+ ********************************************************************************/
+#if defined(CONFIG_ARM_MPU) && (defined(CONFIG_APP_BINARY_SEPARATION) || defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION))
```

Comment:
please recheck the config checks
CONFIG_MPU_STACK_OVERFLOW_PROTECTION already depends on ARM_MPU and BUILD_PROTECTED

Reply:
yes modified